zkevm-circuits/src/circuit_tools/cell_manager.rs

```diff
@@ -70,67 +107,389 @@ impl<F: Field> Expr<F> for &Cell<F> {
     }
 }
 
-/// CellType
-#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub enum CellType {
-    /// General
-    Storage,
+#[derive(Clone, Debug)]
+pub struct CellConfig<C: CellTypeTrait> {
+    cell_type: C,
+    num_columns: usize,
```

Comment:
Multiple columns necessary? This is maybe done to support e.g. lookups without rlc compression?

Reply:
You may or may not do rlc compression with this specific cell type, cuz to the CM a `CellType` is just a label. If you need one column for rlc data only you can do (`CellType::RLC`, 1 col, phase1, not permuted) and perform rlc in your circuit logic.